[TwigBridge] Money type not rendering properly

mogilvie opened the pull request:
Any classes passed to the money input are concatenated to the input-group class, so neither render. Add a space.

| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | 

Money type not working when additional classes are past to the twig form. The additional classes are appended without a space, so the combined class might be "input-groupcol-md-3" and neither get rendered.

mogilvie committed:
Money type not rendering properly.

Any classes passed to the money input are concatenated to the input-group class, so neither render.
Add a space.

94noni commented:
hum in fact, when checking the code, I found other occurrences, seem it has been made voluntarily like this
or otherwise need to update the other files too

derrabus commented:
Does this affect 5.4 as well?

xabbuh commented:
> Does this affect 5.4 as well?

Yes, should be merged into 5.4.

mogilvie committed:
Change class definition to join classess

Previous version introduced a white space character if no classes were passed. Failing tests.

mogilvie committed:
Update bootstrap_5_layout.html.twig

mogilvie committed:
Update bootstrap_5_layout.html.twig

Sorry, twig not my strong suit. Looking to merge the base input-class with an array of additional that may not be defined

mogilvie commented:
Sorry all the tests are wrecking me. Twig not my area of expertise obviously!

xabbuh commented on src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_5_layout.html.twig:
```diff
@@ -6,7 +6,11 @@
+        {% if group_class is defined %}
+            {% set inputClass = group_class|merge(inputClass) %}
+        {% endif %}
+        <div class="{{ inputClass | join(' ')}}">
```
What was wrong with `<div class="input-group {{ group_class|default('') }}">`?

mogilvie replied:
The above renders as ```<div class="input-group ">``` which fails the tests. 
Test requires an exact match with ```<div class="input-group">``` (note the space in the class.)

stof replied:
is this a test shared between different themes ? If no, I suggest amending the test to match the new behavior.

mogilvie replied:
I seem to be coming up against a deprecation issue around the fact that group_class is not defined, or is able to be undefined. So later versions of 8.2 are failing on legacy deprecation notices.
Trying to filter by length or 'is defined' raises a null type error.
Not having much experience with twig, I'm suspecting my solutions are looking hacky!

mogilvie replied:
> is this a test shared between different themes ? If no, I suggest amending the test to match the new behavior.

It seems it's an issue across multiple themes. But that they each have their own block.

stof replied:
My question is not whether the issue is shared between themes (which is a separate question, even though it is an interesting one). My question is whether the tests rejecting your initial patch (adding a space all the time after `input-group`) is shared with other themes or if its specific to this theme. If it is specific to this theme, we could patch the test as well and use the simple patch (which is the simpler code and so my preferred choice)

mogilvie replied:
Tests are specific to the respective theme, I'll revert back to the simpler version and modify the tests. Thanks for the steer on your prefence.

xabbuh replied:
If you switch back to the `<div class="input-group {{ group_class|default('') }}">`, there are two tests in `AbstractBootstrap5LayoutTest` and `FormExtensionBootstrap5LayoutTest` that would need to be adjusted. I think I agree with @stof that doing that change is good enough.

mogilvie committed:
Update bootstrap_5_layout.html.twig

mogilvie committed:
Update bootstrap_5_layout.html.twig

mogilvie committed:
Update bootstrap_5_layout.html.twig

mogilvie committed:
Update bootstrap_5_layout.html.twig

mogilvie committed:
Update bootstrap_5_layout.html.twig

mogilvie committed:
Add space into Money test class

mogilvie committed:
Add space to money widget class

mogilvie committed:
Add space to money widget

mogilvie committed:
Add space to money widget

mogilvie committed:
Space to money type

stof commented:
@mogilvie why closing this PR ?

mogilvie commented:
Was messing it up too much using the github web IDE. Going to do it on a local version in PhpStorm.
